### DISCUSSION TITLE
Flows: Event Hook Filter (Blocking) on create inserting null values

### DISCUSSION THREAD

**Discussion by DanielProp** [directus/directus#15697] [Archive]
I'm trying to create a flow that filters new items in a collection. 
For this I create a flow, with an `Event Hook` trigger, type of `Filter (Blocking)`, scope `items.create`. Whatever I do I can't seem get the trigger payload to be inserted, all values are null(or their default value if set).

I need it to be `Filter (Blocking)` because I'd like to prevent the creation in certain circumstances.

What am I missing? 
Thank you for your time.

**Comment by azrikahar** [comment]
It can depend on the flow setup, but you will need to make sure the actual trigger payload for said Filter hook is being returned by the flow, since that payload is what's being used for the actual creation.

This answer here goes much more in depth on what I described: https://github.com/directus/directus/discussions/15601#discussioncomment-3661117
